artifactory/commands/transferconfig/transferconfigmerge.go

```diff
@@ -0,0 +1,373 @@
+	}
+	log.Info(coreutils.PrintTitle(coreutils.PrintBold("========== Phase 2/3 - Transferring Projects ==========")))
+	if !transferProjects {
+		log.Info(fmt.Sprintf("Your source version is %s, there is no Projects in this version we will transfer your repositories", sourceArtifactoryVersion))
+	} else {
+		sourceServiceAccessManager, err := utils.CreateAccessServiceManager(tcc.sourceServerDetails, tcc.dryRun)
+		sourceProjects, err := sourceServiceAccessManager.GetAllProjects()
+		if err != nil {
+			return err
+		}
+		targetServiceAccessManager, err := utils.CreateAccessServiceManager(tcc.targetServerDetails, tcc.dryRun)
+		targerprojects, err := targetServiceAccessManager.GetAllProjects()
+		if err != nil {
+			return err
+		}
+
+		var projectConflicts []ProjectConflict
+		conflict := false
+		var isConflict bool
+		for _, sourceProject := range sourceProjects {
+			for _, targetProject := range targerprojects {
+
+				projectConflicts, isConflict = tcc.validateProjectConflict(sourceProject, targetProject, projectConflicts)
+				conflict = conflict || isConflict
+			}
+			if !conflict {
+				err := targetServiceAccessManager.CreateProject(accessServices.ProjectParams{ProjectDetails: sourceProject})
+				if err != nil {
+					return err
+				}
+			}
+		}
+		csvPath, err := tcc.createConflictsCSVSummary(projectConflicts, time.Now())
+		if err != nil {
+			log.Error("Couldn't create the long properties CSV file", err)
+			return err
+		}
+
+		log.Info(fmt.Sprintf("Founded %d projectConflicts projects between the source service and the target, check in csv file we created for you in %s", len(projectConflicts), csvPath))
+	}
+	log.Info(coreutils.PrintTitle(coreutils.PrintBold("========== Phase 3/3 - Transferring Repositories ==========")))
+	err = tcc.mergeRepositories(sourceServiceManager, targetServiceManager)
+
+	return err
+}
+
+func (tcc *TransferConfigCommand) validateProjectConflict(sourceProject, targetProject accessServices.Project, conflicts []ProjectConflict) ([]ProjectConflict, bool) {
+	result := false
+	if sourceProject.ProjectKey == targetProject.ProjectKey || sourceProject.DisplayName == targetProject.DisplayName {
+		s := ""
+		result = true
+		if sourceProject.DisplayName != targetProject.DisplayName {
+			s = tcc.addToDifferentProperty(s, "Display Name")
+		}
+		if sourceProject.ProjectKey != targetProject.ProjectKey {
+			s = tcc.addToDifferentProperty(s, "Project Key")
+		}
+		if result {
+			if sourceProject.Description != targetProject.Description {
+				s = tcc.addToDifferentProperty(s, "Description")
+			}
+			if sourceProject.StorageQuotaBytes != targetProject.StorageQuotaBytes {
+				s = tcc.addToDifferentProperty(s, "Storage Quota Bytes")
+			}
+			if sourceProject.SoftLimit != nil && targetProject.SoftLimit != nil {
+				if *sourceProject.SoftLimit != *targetProject.SoftLimit {
+					s = tcc.addToDifferentProperty(s, "Soft Limit")
+				}
+			}
+			if sourceProject.SoftLimit == nil || targetProject.SoftLimit == nil {
```

Comment:
duplicate code

Reply:
It's not duplicated, first we check the case that the softLimit is not nil--> so I check the deep value 
then we check if at least one is nil